src/app/issuer/components/request-badge/request-badge.component.ts

```diff
@@ -76,17 +78,61 @@ export class RequestBadgeComponent extends BaseRoutableComponent implements OnIn
 	badgeClassLoaded: Promise<unknown>;
 	badgeClass: PublicApiBadgeClassWithIssuer;
 
+	qrCodeValid = false;
+	qrCodeExpiredMessage: string;
+	qrCodeData: ApiQRCode;
+
 	get issuerSlug() {
 		return this.route.snapshot.params['issuerSlug'];
 	}
 
 	ngOnInit(): void {
+		this.checkQrCodeValidity();
 		this.requestForm.setValue({
 			...this.requestForm.value,
 			qrCodeId: this.qrSlug,
 		});
 	}
 
+	private checkQrCodeValidity(): void {
+		if (!this.qrSlug) {
+			this.qrCodeValid = false;
+			return;
+		}
+
+		this.publicApiService
+			.getQrCode(this.qrSlug)
+			.then((qrCode: ApiQRCode) => {
+				this.qrCodeData = qrCode;
+				this.qrCodeValid = this.isQrCodeValid(qrCode);
+
+				if (!this.qrCodeValid) {
+					this.qrCodeExpiredMessage = this.translate.instant('QrCode.expired');
+				}
+			})
+			.catch((error) => {
+				console.error('Error fetching QR code:', error);
+				this.qrCodeValid = false;
+				this.qrCodeExpiredMessage = "this.translate.instant('General.notFound')";
+			});
+	}
+
+	private isQrCodeValid(qrCode: ApiQRCode): boolean {
+		// If no validity dates set, consider it always valid
+		if (!qrCode.valid_from || !qrCode.expires_at) {
+			return true;
+		}
+
+		const now = new Date();
+		const validFrom = new Date(qrCode.valid_from);
+		const expiresAt = new Date(qrCode.expires_at);
+
+		const todayStart = new Date(now);
+		todayStart.setHours(0, 0, 0, 0);
+
+		return validFrom <= now && expiresAt >= todayStart;
+	}
+
```

Comment:
This works but has some minor UX issues: Can you add a loading indicator so people on slow connections don't get confused. They would first see "expired qr code" before the call to the public api resolves when they would see the form to request the badge.